### PR TITLE
Document how to connect to Cloud SQL

### LINKED ISSUE #9985: Driver does not support socket connection
**Describe the bug**
I try to connect a projet, build in native mode and deploy on google cloud run, to google sql.

**Expected behavior**
My app cannot reach google sql

**Actual behavior**
I get an error : 
`ERROR [org.hib.eng.jdb.spi.SqlExceptionHelper] (main) Driver does not support the provided URL: jdbc:postgres://[user]:[pwd]@/[db-name]?unix_sock=/cloudsql/[db-connection-name]"
`
**To Reproduce**
1. Build image in native mode
2. deploy on google run

**Configuration**
Pom.xml
```
  <properties>
    <quarkus-plugin.version>1.5.1.Final</quarkus-plugin.version>
    <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
    <quarkus.platform.group-id>io.quarkus</quarkus.platform.group-id>
    <quarkus.platform.version>1.5.1.Final</quarkus.platform.version>
	<testcontainers.version>1.14.1</testcontainers.version>
    <surefire-plugin.version>2.22.1</surefire-plugin.version>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
	<docker-plugin.version>0.28.0</docker-plugin.version>
    <maven.compiler.source>1.8</maven.compiler.source>
    <maven.compiler.target>1.8</maven.compiler.target>
  </properties>

	<dependency>
		<groupId>io.quarkus</groupId>
		<artifactId>quarkus-hibernate-orm-panache</artifactId>
	</dependency>	
	<dependency>
		<groupId>io.quarkus</groupId>
		<artifactId>quarkus-jdbc-postgresql</artifactId>
	</dependency>   
	<!--dependency>
		<groupId>io.quarkus</groupId>
		<artifactId>quarkus-reactive-pg-client</artifactId>
	</dependency-->	
	<dependency>
		<groupId>com.google.cloud.sql</groupId>
		<artifactId>postgres-socket-factory</artifactId>
		<version>1.0.16</version>
	</dependency>	
```

application.properties
```
quarkus.datasource.db-kind=postgresql
quarkus.datasource.jdbc.url=jdbc:postgres://${DATASOURCE_USER}:${DATASOURCE_PWD}@/${DATASOURCE_DBNAME}?unix_sock=/cloudsql/${DATASOURCE_URL}
```
Where variables are environnements variables on google cloud run

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
 - Output of `java -version`: 
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 1.5.1.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 

**Additional context**
Dockerfile.multistage

```
## Stage 1 : build with maven builder image with native capabilities
FROM quay.io/quarkus/ubi-quarkus-native-image:19.3.1-java11 as nativebuilder
RUN mkdir -p /tmp/ssl \
  && cp /opt/graalvm/lib/security/cacerts /tmp/ssl/
  
FROM quay.io/quarkus/centos-quarkus-maven:19.3.1-java11 AS build
RUN mkdir -p /tmp/ssl \
  && cp /opt/graalvm/lib/security/cacerts /tmp/ssl/
COPY pom.xml /usr/src/app/
RUN mvn -f /usr/src/app/pom.xml -B de.qaware.maven:go-offline-maven-plugin:1.2.5:resolve-dependencies
COPY src /usr/src/app/src
USER root
RUN chown -R quarkus /usr/src/app
USER quarkus
RUN mvn -f /usr/src/app/pom.xml -Pnative clean package -DskipTests=true

## Stage 2 : create the docker final image
FROM registry.access.redhat.com/ubi8/ubi-minimal
WORKDIR /work/
COPY --from=nativebuilder /tmp/ssl/ /work/
COPY --from=build /usr/src/app/target/*-runner /work/application

# set up permissions for user `1001`
RUN chmod 775 /work /work/application \
  && chown -R 1001 /work \
  && chmod -R "g+rwX" /work \
  && chown -R 1001:root /work

EXPOSE 8080
USER 1001
ENV DISABLE_SIGNAL_HANDLERS true
CMD ["./application", "-Dquarkus.http.host=0.0.0.0", "-Djavax.net.ssl.trustStore=/work/cacerts"]

```

Note : find this feature : https://github.com/jtama-op/quarkus/tree/18280ee018d82acef601fc022a5c6923bfb7a451/extensions/agroal

### PR DESCRIPTION
**Pull request by loicmathieu**
Fixes #9985

Draft PR due to #15782 that prevent Cloud SQL to work on dev mode.

### CODE REVIEW

File: docs/src/main/asciidoc/deploying-to-google-cloud.adoc [hunk 1]
```diff
@@ -217,9 +218,51 @@ After you answer to these questions, it will deploy your application.
 
 When the deployment is done, the output will display the URL to access your application.
 
+== Using Cloud SQL
+
+Google Cloud SQL provides managed instances for MySQL, Postgres and Sqlserver.
+Quarkus has support for all three databases.
+
+To make them works with Cloud SQL you first need to use the corresponding JDBC extension, for example for Postgres,
```
Comment (geoand) [inline docs/src/main/asciidoc/deploying-to-google-cloud.adoc:226]:
```suggestion
To make them work with Cloud SQL, you first need to use the corresponding JDBC extension, for example for PostgreSQL,
```

File: docs/src/main/asciidoc/deploying-to-google-cloud.adoc [hunk 2]
```diff
@@ -217,9 +218,51 @@ After you answer to these questions, it will deploy your application.
 
 When the deployment is done, the output will display the URL to access your application.
 
+== Using Cloud SQL
+
+Google Cloud SQL provides managed instances for MySQL, Postgres and Sqlserver.
```
Comment (geoand) [inline docs/src/main/asciidoc/deploying-to-google-cloud.adoc:223]:
```suggestion
Google Cloud SQL provides managed instances for MySQL, PostgreSQL and SQLServer.
```

**Comment by loicmathieu** [comment]
As https://github.com/quarkusio/quarkus/issues/15782 didn't seems to be an easy one (no fix possible on our side), maybe I can proceed with this PR and add a link to this issue saying that dev mode didn't work with socket factory at the moment.

I wonder if it works for mysql/sqlserver, didn't test it.

I wonder also if a link to this section could be added to the datasource link and if so in wich section ?

**Comment by geoand** [comment]
> As #15782 didn't seems to be an easy one (no fix possible on our side), maybe I can proceed with this PR and add a link to this issue saying that dev mode didn't work with socket factory at the moment.
> 

Well, that's not entirely true :). We can on our side transform the class in question using an ASM transformer and force the use of the TCCL. 
It's not a nice solution, but you are most welcome to try it until we have a proper upstream fix.

**Review by geoand — APPROVED** [review]

**Comment by gsmet** [comment]
Rebased, force-pushed and added a commit with some fixes.

**Pull request merged**